### PR TITLE
docs: update vehicle interface with some samples

### PR DESCRIPTION
**Pull request by ismetatabay**
## Description

This PR updates vehicle interface documentation with more information.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by ismetatabay** [comment]
@mitsudome-r, Hi mitsudome-san this PR is ready for review.

File: docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/.pages [hunk 1]
```diff
@@ -1,3 +1,5 @@
 nav:
-  - Creating a vehicle interface for an Ackermann kinematic model: creating-a-vehicle-interface-for-an-ackermann-kinematic-model.md
+  - vehicle-interface.md
```
Comment (mitsudome-r) [inline docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/.pages:2]: Maybe it's better to have page title more obvious
```suggestion
  - Vehicle Interface Overview: vehicle-interface.md
```

File: docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/vehicle-interface.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Vehicle interface
+
+## What is the vehicle interface?
+
+The purpose of the vehicle interface package is to convert the control messages calculated from the autoware into a form that the vehicle can understand and transmit them to the vehicle (CAN, Serial message, etc.) and to decode the data coming from the vehicle and publish it through the ROS 2 topics that the autoware expects.
+So, we can say that the purposes of the vehicle_interface are:
+
+1. Converting Autoware control commands to a vehicle-specific format. Example control commands of autoware:
```
Comment (mitsudome-r) [inline docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/vehicle-interface.md:8]: This line ends with `Example control commands of autoware:`, but it does't have provide the example in the next line.

File: docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/vehicle-interface.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+So think of the vehicle interface as a module that runs the vehicle's control device to realize the input commands provided by Autoware.
+
+<figure markdown>
+  ![vehicle_interface_IO](images/Vehicle-Interface-Bus-ODD-Architecture.drawio.svg){ align=center }
+  <figcaption>
+    An example of inputs and outputs for vehicle interface
+  </figcaption>
+</figure>
+
+There are two types of controlling your own vehicle:
```
Comment (mitsudome-r) [inline docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/vehicle-interface.md:34]: I found the explanation of the two types hard to follow. Could we set a call to get aligned on what you are trying to say here?

File: docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/creating-vehicle-interface.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+  ...
+}
+
+```
+
+- Communication between the vehicle interface and your vehicle's control device
+
+  - If you are planning to drive by autoware with your vehicle, then your vehicle must satisfy some requirements:
+
+    | Type A                                                                                                                                         | Type B                                                                                                                                         |
```
Comment (mitsudome-r) [inline docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/creating-vehicle-interface.md:159]: I believe Type A and Type B refers to the two types that are mentioned in `vehicle-interface.md` page.
Maybe we should consider either moving this table to `vehicle-interface` or move the explanation about the two types in `vehicle-interface.md` into this file.

File: docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/.pages (hunk 1 above)
Comment (ismetatabay) [reply docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/.pages]: Done :+1:

File: docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/vehicle-interface.md (hunk 2 above)
Comment (ismetatabay) [reply docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/vehicle-interface.md:8]: done :+1:

**Comment by ismetatabay** [comment]
Hi @mitsudome-r san, thank you for valuable comments. I updated   document according to the your reviews :+1:

**Review by mitsudome-r — APPROVED** [review]
